lib/zipkin-tracer/excon/zipkin-tracer.rb

```diff
@@ -0,0 +1,85 @@
+require 'finagle-thrift/trace'
+require 'finagle-thrift/tracer'
+require 'uri'
+require 'excon'
+
+module ZipkinTracer
+  class ExconHandler < Excon::Middleware::Base
+    def initialize(_)
+      super
+    end
+
+    def error_call(datum)
+      super(datum)
+    end
+
+    def request_call(datum)
+      trace_id = TraceGenerator.new.next_trace_id
+
+      TraceContainer.with_trace_id(trace_id) do
+        b3_headers.each do |method, header|
+          datum[:headers][header] = trace_id.send(method).to_s
+        end
+
+        trace!(datum, trace_id) if trace_id.sampled?
+      end
+
+      super(datum)
+    end
+
+    def response_call(datum)
```

Comment:
if there's some sort of unhandled exception, you can add a binary annotation of type string with key "error" with the cause. This will make it show up in the UI in red.